dashboard/test/models/coteacher_test.rb

```diff
@@ -0,0 +1,30 @@
+require 'test_helper'
+
+class CoteacherTest < ActiveSupport::TestCase
+  test "user must be teacher" do
+    teacher = create(:teacher)
+    student = create(:student)
+
+    teacher_section = Section.create(users: [teacher], name: "a section")
+    assert teacher_section.persisted?
+
+    student_section = Section.create(users: [student], name: "a section")
+
+    assert !student_section.persisted?
+    assert_equal ["User must be a teacher"], student_section.coteachers.first.errors.full_messages
+  end
```

Comment:
nit: This is two tests, a test for "teachers can own sections" and a test for "students cannot own sections".